src/rust/plugin-registry/src/server.rs

```diff
@@ -125,16 +187,33 @@ impl PluginRegistryService for PluginRegistry {
         todo!()
     }
 
+    #[tracing::instrument(skip(self, request), err)]
     async fn get_generators_for_event_source(
         &self,
-        _request: Request<GetGeneratorsForEventSourceRequestProto>,
+        request: Request<GetGeneratorsForEventSourceRequestProto>,
     ) -> Result<Response<GetGeneratorsForEventSourceResponseProto>, Status> {
-        // Stub implementation, for integration test smoke-test purposes.
-        // Replace with a real implementation soon!
-        let message = _request.into_inner();
-        Ok(Response::new(GetGeneratorsForEventSourceResponseProto {
-            plugin_ids: [message.event_source_id].into_iter().flatten().collect(),
-        }))
+        let request = request.into_inner();
+        let request =
+            GetGeneratorsForEventSourceRequest::try_from(request).expect("Invalid message");
+
+        match self.get_generators_for_event_source(request).await {
+            Ok(response) => {
+                tracing::debug!(
+                    message="Successfully retrieved generators for event source",
+                    plugin_ids=?response.plugin_ids,
+                );
+                Ok(Response::new(
+                    GetGeneratorsForEventSourceResponseProto::from(response),
+                ))
```

Comment:
🤔 This might just be because this is my first time looking at this code, but we're dealing with _three_ different things called response in this one line of code.

Rather than importing the various Tonic types directly into this module, it would be a bit more clear to just `use tonic;`, and then we can have `tonic::Response::new(...)`.

The way this code is written, a casual reading would imply that `Response` is some type that _we_ created in this crate, which is probably why my brain broke trying to figure out the logical relationship between the three responses here.